haystack/nodes/retriever/dense.py

```diff
@@ -1072,11 +1072,18 @@ def retrieve_batch(
         if scale_score is None:
             scale_score = self.scale_score
 
-        query_embs: List[np.ndarray] = []
-        for batch in self._get_batches(queries=queries, batch_size=batch_size):
-            query_embs.extend(self.embed_queries(queries=batch))
+        # embed_queries is already batched within by batch_size
+        query_embs: np.ndarray = self.embed_queries(queries=queries)
+        batched_query_embs: List[np.ndarray] = []
+        for i in range(0, len(query_embs), batch_size):
+            batched_query_embs.extend(query_embs[i : i + batch_size])
```

Comment:
If I did not misunderstand something in the steps above, this change can get much simpler:

```suggestion
        # embed_queries is already batched within by batch_size
        query_embs = self.embed_queries(queries=queries)
```

Then replace `batched_query_embs` with `query_embs` and it should work.

Same for the other occurrence below.

Reply:
I agree this approach would also work. However, we do want to preserve the batches still so `query_by_embedding_batch` will only query one batch at a time. So thanks for catching the error above with the `extend`!